otherlibs/systhreads/thread.mli

```diff
@@ -42,10 +42,21 @@ val id : t -> int
    It can be used to build data structures indexed by threads. *)
 
 val exit : unit -> unit
-(** Terminate prematurely the currently executing thread. *)
+  [@@ocaml.deprecated "Use a dedicated exception."]
+(** @deprecated In order to guarantee the correct reclamation of
+  resources in calling code, use a dedicated exception to interrupt
+  the currently running thread.
+  The implementation of [exit] has been changed to using such an
+  exception, which is handled in [create]. This should not be relied
+  on.
+
+  Terminate prematurely the currently executing thread. *)
 
 val kill : t -> unit
-(** Terminate prematurely the thread whose handle is given. *)
+  [@@ocaml.deprecated "This function is currently unimplemented."]
```

Comment:
If there are no more implementation of `Thread.kill` around (after removal of vmthreads I imagine), then you know that no code relies on it and it can safely be removed. Or are there other adverse effects I did not think about?

Reply:
It turns out that there is some dead code that refers to it, but which is never called in actual use cases (cf bigbro). So, I prefer deprecation rather than removal: it has low cost, and still makes possible to compile these pieces of code.

Reply:
Makes sense